File: frontend-server/src/app/model/dto/united/calendar-data-dto.ts
import { EmployeeScheduleDTO } from "../employee-schedule-dto";
import { WorkingNorm } from "../../working-norm";
import { SpecialCalendarDate } from "../../special-calendar-date";

export class CalendarDataDTO {
  schedule: EmployeeScheduleDTO[];
  workingNorms: WorkingNorm[];
  specialCalendarDates: SpecialCalendarDate[];
}

File: work-schedule-service/src/main/java/com/averude/uksatse/scheduler/microservice/workschedule/model/CalendarDataDTO.java
package com.averude.uksatse.scheduler.microservice.workschedule.model;

import com.averude.uksatse.scheduler.core.model.dto.EmployeeScheduleDTO;
import com.averude.uksatse.scheduler.core.model.entity.SpecialCalendarDate;
import com.averude.uksatse.scheduler.core.model.entity.WorkingNorm;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@RequiredArgsConstructor
public class CalendarDataDTO {
    private final List<EmployeeScheduleDTO> schedule;
    private final List<WorkingNorm> workingNorms;
    private final List<SpecialCalendarDate> specialCalendarDates;
}
